clang_build/target.py

```diff
@@ -163,47 +159,65 @@ def parse_flags_options(self, options, flags_kind='flags'):
 
-        if 'osx' in options and _platform.PLATFORM == 'osx':
-            flags_dicts.append(options['osx'].get(flags_kind, {}))
-        if 'windows' in options and _platform.PLATFORM == 'windows':
-            flags_dicts.append(options['windows'].get(flags_kind, {}))
-        if 'linux' in options and _platform.PLATFORM == 'linux':
-            flags_dicts.append(options['linux'].get(flags_kind, {}))
+        flags_dicts.append(options.get(_platform.PLATFORM, {}).get(flags_kind, {}))
 
         for fdict in flags_dicts:
             compile_flags     += fdict.get('compile', [])
             link_flags        += fdict.get('link', [])
 
-            if self.build_type == _BuildType.Release:
-                compile_flags += fdict.get('compile_release', [])
-
-            elif self.build_type == _BuildType.Debug:
-                compile_flags += fdict.get('compile_debug', [])
-
-            elif self.build_type == _BuildType.RelWithDebInfo:
-                compile_flags += fdict.get('compile_relwithdebinfo', [])
-
-            elif self.build_type == _BuildType.Coverage:
-                compile_flags += fdict.get('compile_coverage', [])
+            if self.build_type != _BuildType.Default:
+                compile_flags += fdict.get(f'compile_{self.build_type}', [])
 
         return compile_flags, link_flags
 
+    @abstractmethod
     def link(self):
-        # Subclasses must implement
-        raise NotImplementedError()
+        pass
 
+    @abstractmethod
     def compile(self, process_pool, progress_disabled):
-        # Subclasses must implement
-        raise NotImplementedError()
+        pass
 
 
 class HeaderOnly(Target):
+    def __init__(self,
+            project_identifier,
+            name,
+            root_directory,
+            build_directory,
+            headers,
+            include_directories,
+            include_directories_public,
+            build_type,
+            clang,
+            clangpp,
+            options=None,
+            dependencies=None):
+        super().__init__(
+            project_identifier=project_identifier,
+            name=name,
+            root_directory=root_directory,
+            build_directory=build_directory,
+            headers=headers,
+            include_directories=[],
+            include_directories_public=include_directories_public+include_directories,
+            build_type=build_type,
+            clang=clang,
+            clangpp=clangpp,
+            options=options,
+            dependencies=dependencies)
+
     def link(self):
         _LOGGER.info(f'[{self.identifier}]: Header-only target does not require linking.')
 
     def compile(self, process_pool, progress_disabled):
         _LOGGER.info(f'[{self.identifier}]: Header-only target does not require compiling.')
 
+    def add_target_flags(self, target):
+        self.apply_public_flags(target)
+        self.forward_public_flags(target)
+        self.forward_interface_flags(target)
```

Comment:
This is a nitpick, but `apply_public_flags` will not have any effect on a Header only target.